java-grupa8/Dealership/src/main/java/ro/scii/java8/Customer.java

```diff
@@ -0,0 +1,395 @@
+                            "\t New: " + v.isNew() + "\t Price: " + v.getPrice() + " Euro" + "\t Stock No.: " + v.getStockNr() + "\n");
+                }
+            } else if (verifyFilterPSF.equals("n")) {
+                for (ElectricVehicle v : getVehiclesInStockList(list)) {
+                    System.out.println(v.getManufacturer() + " " + v.getVehicleModel() + "\n Power: " + v.getMotorPower() + " kW " +
+                            "\t Range: " + v.getRange() + " km " + "\t Fast Charging: " + v.isHasFastCharging() +
+                            "\t New: " + v.isNew() + "\t Price: " + v.getPrice() + " Euro" + "\t Stock No.: " + v.getStockNr() + "\n");
+                }
+            } else {
+                throw new IllegalArgumentException("Only y or n options can be used.");
+            }
+        } else if (verifyFilterPS.equals("n")) {
+            System.out.println("Do you want to exclude the vehicles that do not have fast charging? y/n");
+            String verifyFilterPF = s.next();
+            if (verifyFilterPF.equals("y")) {
+                for (ElectricVehicle v : getFastChargingList(list))
+                    System.out.println(v.getManufacturer() + " " + v.getVehicleModel() + "\n Power: " + v.getMotorPower() + " kW " +
+                            "\t Range: " + v.getRange() + " km " + "\t Fast Charging: " + v.isHasFastCharging() +
+                            "\t New: " + v.isNew() + "\t Price: " + v.getPrice() + " Euro" + "\t Stock No.: " + v.getStockNr() + "\n");
+            } else if (verifyFilterPF.equals("n")) {
+                for (ElectricVehicle v : list) {
+                    System.out.println(v.getManufacturer() + " " + v.getVehicleModel() + "\n Power: " + v.getMotorPower() + " kW " +
+                            "\t Range: " + v.getRange() + " km " + "\t Fast Charging: " + v.isHasFastCharging() +
+                            "\t New: " + v.isNew() + "\t Price: " + v.getPrice() + " Euro" + "\t Stock No.: " + v.getStockNr() + "\n");
+                }
+            } else {
+                throw new IllegalArgumentException("Only \"y\" or \"n\" options can be used.");
+            }
+        } else {
+            throw new IllegalArgumentException("Only \"y\" or \"n\" options can be used.");
+        }
+
+        System.out.println("Do you want to buy a car? y/n");
+        String buy = s.next();
+        if (buy.equals("y")) {
+            chooseVehicle();
+        } else if (buy.equals("n")) {
+            System.out.println("Thank you for visiting!");
+        } else {
+            throw new IllegalArgumentException("Only \"y\" or \"n\" options can be used.");
+        }
+
+    }
+
+
+    /**
+     * This method allows the customer to choose a vehicle to purchase
+     *
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    public void chooseVehicle() throws IOException, ClassNotFoundException {
```

Comment:
This method does not belong to the customer. The customer might be calling it on a dealership.